### PR TITLE
Improved send_file.rs example to use streaming.

### PR DESCRIPTION
**Pull request by acedogblast**
This is to show what I have wrote so far. It does not compile in the current state.
This addresses issues #3432

### CODE REVIEW

File: examples/send_file.rs [hunk 1]
```diff
@@ -41,7 +44,7 @@ async fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
     }
 }
 
-async fn response_examples(req: Request<hyper::body::Incoming>) -> Result<Response<Full<Bytes>>> {
+async fn response_examples(req: Request<hyper::body::Incoming>) -> Result<Response<BoxBody<Bytes, std::io::Error>>> {
```
Comment (seanmonstar) [inline examples/send_file.rs:47]:
```suggestion
async fn response_examples(
    req: Request<hyper::body::Incoming>,
) -> Result<Response<BoxBody<Bytes, std::io::Error>>>
```

File: examples/send_file.rs [hunk 2]
```diff
@@ -53,18 +56,35 @@ async fn response_examples(req: Request<hyper::body::Incoming>) -> Result<Respon
 }
 
 /// HTTP status code 404
-fn not_found() -> Response<Full<Bytes>> {
+fn not_found() -> Response<BoxBody<Bytes, std::io::Error>> {
     Response::builder()
         .status(StatusCode::NOT_FOUND)
-        .body(Full::new(NOTFOUND.into()))
+        .body(Full::new(NOTFOUND.into().boxed()))
```
Comment (seanmonstar) [inline examples/send_file.rs:62]:
```suggestion
        .body(Full::new(NOTFOUND.into()).boxed())
```

**Review by seanmonstar — COMMENTED** [review]
So far I noticed these two changes.

**Comment by acedogblast** [comment]
We still have build errors:
line 82 mismatched types
expected `StreamBody<_>`, found `ReaderStream<File>`

and line 61 mismatched types
expected `Response<BoxBody<Bytes, Error>>`, found `Response<BoxBody<_, Infallible>>`

File: examples/send_file.rs [hunk 3]
```diff
@@ -41,7 +44,9 @@ async fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
     }
 }
 
-async fn response_examples(req: Request<hyper::body::Incoming>) -> Result<Response<Full<Bytes>>> {
+async fn response_examples(
+    req: Request<hyper::body::Incoming>,
+) -> Result<Response<BoxBody<Bytes, std::io::Error>>>
```
Comment (seanmonstar) [inline examples/send_file.rs:49]:
```suggestion
) -> Result<Response<BoxBody<Bytes, std::io::Error>>> {
```

**Comment by seanmonstar** [comment]
You'll need to use `map_err(|e| e as _)` on the `Full`, and you'll need to import `StreamExt` and use `.map_ok(Frame::data)` on the `ReaderStream` before it can be converted into a `StreamBody`.

**Comment by acedogblast** [comment]
I am not sure what you mean by `map_err(|e| e as _)` on the `Full` .
Also are you referring to `StreamExt` from the tokio-stream crate(https://docs.rs/tokio-stream/latest/tokio_stream/trait.StreamExt.html)?

File: examples/send_file.rs [hunk 4]
```diff
@@ -53,18 +58,35 @@ async fn response_examples(req: Request<hyper::body::Incoming>) -> Result<Respon
...
     }
 
-    Ok(not_found())
+    let file: File = file.unwrap();
+
+    // Wrap to a tokio_util::io::ReaderStream
+    let reader_stream = ReaderStream::new(file);
+
+    // Convert to http_body_util::BoxBody
+    let stream_body = StreamBody::from(reader_stream);
```
Comment (seanmonstar) [inline examples/send_file.rs:82]:
```suggestion
    let stream_body = StreamBody::new(reader_stream.map_ok(Frame::data));
```

File: examples/send_file.rs [hunk 5]
```diff
@@ -53,18 +58,35 @@ async fn response_examples(req: Request<hyper::body::Incoming>) -> Result<Respon
 }
 
 /// HTTP status code 404
-fn not_found() -> Response<Full<Bytes>> {
+fn not_found() -> Response<BoxBody<Bytes, std::io::Error>> {
     Response::builder()
         .status(StatusCode::NOT_FOUND)
-        .body(Full::new(NOTFOUND.into()))
+        .body(Full::new(NOTFOUND.into()).boxed())
```
Comment (seanmonstar) [inline examples/send_file.rs:64]:
```suggestion
        .body(Full::new(NOTFOUND.into()).map_err(|e| match e {}).boxed())
```

File: examples/send_file.rs [hunk 6]
```diff
@@ -3,10 +3,13 @@
 use std::net::SocketAddr;
 
 use hyper::server::conn::http1;
-use tokio::net::TcpListener;
-
+use tokio::{
+    net::TcpListener,
+    fs::File,
+};
+use tokio_util::io::ReaderStream;
```
Comment (seanmonstar) [inline examples/send_file.rs:10]:
```suggestion
use tokio_util::io::ReaderStream;
use futures_util::TryStreamExt;
use hyper::body::Frame;
```

**Review by seanmonstar — COMMENTED** [review]
I've tested out all the rest of the changes I'm suggested, and it works at the end.

**Comment by acedogblast** [comment]
I can also confirm that it builds and works. Now ready to pull.

**Pull request merged**